src/firstboot/firstboot.c

```diff
@@ -308,28 +308,37 @@ static int prompt_locale(void) {
         return 0;
 }
 
-static int process_locale(void) {
-        const char *etc_localeconf;
+static int process_locale(int rfd) {
+        _cleanup_free_ char *p = NULL;
+        _cleanup_close_ int pfd = -EBADF;
         char* locales[3];
         unsigned i = 0;
         int r;
 
-        etc_localeconf = prefix_roota(arg_root, "/etc/locale.conf");
-        if (laccess(etc_localeconf, F_OK) >= 0 && !arg_force) {
-                log_debug("Found %s, assuming locale information has been configured.",
-                          etc_localeconf);
+        r = chase_symlinks_at(rfd, "etc", CHASE_AT_RESOLVE_IN_ROOT|CHASE_NONEXISTENT, &p, &pfd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to resolve %s/etc/: %m", FORMAT_FD_PATH(rfd));
+
+        if (faccessat(pfd, "locale.conf", F_OK, AT_SYMLINK_NOFOLLOW) >= 0 && !arg_force) {
```

Comment:
hmm, pfd might refer to /etc, or to /, you never know, right? that seems wrong thne?

Reply:
If I understand chase_symlinks() correctly, `pfd` will either refer to /etc or be invalid